internal/vehicle/ford.go

```diff
@@ -214,43 +224,40 @@ func (v *Ford) vehicleStatus() (fordVehicleStatus, error) {
 
 		if elapsed := time.Since(lastUpdate); err == nil && elapsed > fordOutdatedAfter {
 			v.log.DEBUG.Printf("vehicle status is outdated (age %v > %v), requesting refresh", elapsed, fordOutdatedAfter)
-			res, err = v.vehicleStatusRefresh()
+
+			if err = v.refreshRequest(); err == nil {
+				err = api.ErrMustRetry
+			}
 		}
 	}
 
 	return res, err
 }
 
-// vehicleStatusRefresh triggers an update and waits until refreshed data is available or request times out
-func (v *Ford) vehicleStatusRefresh() (fordVehicleStatus, error) {
-	commandId, err := v.requestRefresh()
+// refreshResult triggers an update if not already in progress, otherwise gets result
+func (v *Ford) refreshResult() (res fordVehicleStatus, err error) {
+	uri := fmt.Sprintf("%s/api/vehicles/v3/%s/statusrefresh/%s", fordAPI, v.vin, v.refreshId)
 
-	var res fordVehicleStatus
-	if err == nil {
-		uri := fmt.Sprintf("%s/api/vehicles/v3/%s/statusrefresh/%s", fordAPI, v.vin, commandId)
-
-		// if status attribute in JSON response is 200, update is complete, otherwise server is still
-		// waiting for vehicle and the request needs to be repeated
-		for counter := 0; counter < fordMaxRefreshTrials && res.Status != 200 && err == nil; counter++ {
-			var req *http.Request
-			if req, err = v.request(http.MethodGet, uri); err == nil {
-				err = v.DoJSON(req, &res)
-			}
-
-			time.Sleep(1500 * time.Millisecond)
-		}
+	var req *http.Request
+	if req, err = v.request(http.MethodGet, uri); err == nil {
+		err = v.DoJSON(req, &res)
+	}
 
-		if err == nil && res.Status != 200 {
-			err = fmt.Errorf("refresh failed: status %d", res.Status)
+	// if status attribute in JSON response is 200, update is complete, otherwise server is still
+	// waiting for vehicle and the request needs to be repeated
+	if err == nil {
```

Comment:
v.refreshId muss auch dann zurückgesetzt werden wenn err != nil && err != err.MustRefresh

Reply:
Das ist ne gute Frage. Könnte bei der nächsten Iteration ja klappen (Netzwerkproblem)? Aktuell würde das weiter passieren bis nachgelagert ErrTimeout kommt. Was meinst Du?

Reply:
Guter Punkt, dann muss aber die Bedingung für den Timeout geändert werden (Zeile 206).
`if res, err = v.refreshResult(); err != nil && time.Since(v.refreshTime) > fordRefreshTimeout {`

Reply:
willst Du's Dir nochmal greifen?

Reply:
ich kann in dein Repo nicht pushen - für die eine Zeile rentiert sich nicht einen neuen Pull-Request zu öffnen (oder es gibt einen Trick, den ich nicht kenne)

Reply:
Aktualisiert- hoffe ich hab das getroffen was Du meintest

Reply:
Leider nicht ganz, so meinte ich das:
https://github.com/MrJayC/evcc/commit/2ebb551a8de98f47f87b59d657df81c64021510a

Habe das jetzt so laufen, hat heute mit dem Laden nicht mehr geklappt, aber morgen werde ich testen.

Reply:
Das wollte ich eigentlich nicht weil damit ein "echter" Fehler hinter einem ErrTimeout versteckt wird. Am Verhalten sollte es aber nichts ändern? Bitte check nochmal ob das nicht doch so wie hier ok wäre.

Reply:
Funktioniert auch nicht - da ein "echter" Fehler dazu führt, dass der Cache keinen Refresh mehr durchführt, folglich kommt es auch nicht zum Timeout. Lass mich etwas nachdenken...

Reply:
Eigentlich einfach

		if err != nil && time.Since(v.refreshTime) > fordRefreshTimeout {
			v.refreshId = ""
			err = api.ErrTimeout
		}

oder?

Reply:
bzw einfach:

Reply:
auch nciht, aber jetzt hab ich das problem verstanden, mom

Reply:
Ok, neuer Versuch- denke es ist einfacher alle Logik an einem Platz zu haben. Jetzt gibts keine Kopfschmerzen mehr ;)

Reply:
So ist es perfekt 👌 
Ich würde sagen „das ist es“ (auf den PR bezogen). Ich zieh die letzte Version auf jeden Fall nochmal auf mein Produktivsystem zum Test, erwarte aber keine Fehler.

Reply:
Top, dann rein damit!